### PR TITLE
Risen Stocks Frontend Impl

### PR DESCRIPTION
**Pull request by Vexrax**
This adds all the FE components for displaying the value of teams. Theres no Buying/selling yet

## League Page
image

## Team Page
image

## Notes:
- I played around with the kfactor a bit and 150 seems like reasonable amount, anything more and the swings from 1 game were way too wild
- I also added a draft league sheet parser so we can support draft league, teambuilders sheet is a little bit too confusing for me so im dodging it for now

### CODE REVIEW

File: BackEnd/src/business/sheets/draftParser.ts [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+      teamName: row[0],
+      abrv: row[1],
+      win: row[2],
+      loss: row[3],
+      top: getPlayerIdentifierFromString(row[7]),
+      jungle: getPlayerIdentifierFromString(row[9]),
+      mid: getPlayerIdentifierFromString(row[11]),
+      adc: getPlayerIdentifierFromString(row[13]),
+      support: getPlayerIdentifierFromString(row[15]),
+      sub1: undefined, // No defined subs in draft league
```
Comment (afieldi) [inline BackEnd/src/business/sheets/draftParser.ts:15]: What about the rest of the leagues lol

File: BackEnd/src/business/sheets/draftParser.ts [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+      sub2: undefined,
+      sub3: undefined,
+      sub4: undefined,
+      sub5: undefined,
+    };
   }
 
   public isValidRow(data: string[]): boolean {
-    return false;
+    return data.length > 5 && this.isValidRowData(data[0]) && this.isValidRowData(data[1]) && hasValidOPGG(data[17]);
```
Comment (afieldi) [inline BackEnd/src/business/sheets/draftParser.ts:24]: I'm reading opgg's on 15 https://docs.google.com/spreadsheets/d/1Af6MlaxpZkKWokvsi1cmBNl1L0tkP2Ld8aow9s45U2g/edit?gid=736224548#gid=736224548

File: BackEnd/src/business/sheets/draftParser.ts (hunk 1 above)
Comment (Vexrax) [reply BackEnd/src/business/sheets/draftParser.ts:15]: Premade is covered via the premadeParser, this is the draft league parser which is a diff format

File: BackEnd/src/business/sheets/draftParser.ts (hunk 2 above)
Comment (Vexrax) [reply BackEnd/src/business/sheets/draftParser.ts:24]: Thats a premade sheet, this is a draft league parser

File: Common/Interface/Internal/stocks.ts [hunk 3]
```diff
@@ -33,10 +33,10 @@ export interface GetPortfolioResponse {
 }
 
 export interface GetStockTimelineResponse {
-    timeline: Map<number, StockTimelineEntry[]>
+    timeline: { [k: number]: StockTimelineEntry[]; }
```
Comment (afieldi) [inline Common/Interface/Internal/stocks.ts:36]: Personally I usually just use `Record<number, StockTimelineEntry[]>`

File: front-end/src/components/charts/stock-timeline-chart.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  // Set saturation to 70% and lightness to 50% for good contrast with white text
+  const saturation = 70;
+  const lightness = 50;
+
+  // Convert HSL to RGB and then to hexadecimal color code
+  return hslToHex(hue, saturation, lightness);
+}
+
+// Helper function to convert HSL to Hex
+function hslToHex(h: number, s: number, l: number): string {
```
Comment (afieldi) [inline front-end/src/components/charts/stock-timeline-chart.tsx:103]: Imma just trust this makes sense

Comment (Vexrax) [reply front-end/src/components/charts/stock-timeline-chart.tsx:103]: TLDR: this is code makes sure that there is good contract in the colors selected.

File: Common/Interface/Internal/stocks.ts (hunk 3 above)
Comment (Vexrax) [reply Common/Interface/Internal/stocks.ts:36]: Updated

**Review by afieldi — APPROVED** [review]

**Pull request merged**